### PR TITLE
"Deprecation" for onerror

### PR DESCRIPTION
**Pull request by timholy**
Plus misc changes that eliminate warnings on 0.6. Leaves julia 0.4 in the dust.

### CODE REVIEW

File: src/core.jl [hunk 1]
```diff
@@ -187,10 +187,14 @@ const _messages = Channel{Nullable{Message}}(CHANNEL_SIZE)
...
-and a `CapturedException` with the fields `ex` which is the original exception
-object, and `processed_bt` which is the backtrace of the exception.
+The third (optional) argument, `onerror`, is a callback triggered when
+the update ends in an error. The callback receives 4 arguments,
+`onerror(sig, val, node, capex)`, where `sig` and `val` are the Signal
+and value that `push!` was called with, respectively, `node` is the
+Signal whose action triggered the error, and `capex` is a
+`CapturedException` with the fields `ex` which is the original
+exception object, and `processed_bt` which is the backtrace of the
+exception.
```
Comment (JobJob) [inline src/core.jl]: Well phrased

**Comment by timholy** [comment]
@shashi, are you OK with abandoning 0.4?

**Comment by timholy** [comment]
Thanks for the review, @JobJob! Appreciate all you've done here.

**Comment by shashi** [comment]
Yes, fine with dropping 0.4. Cheers!

**Pull request merged**